Lib/test/test_concurrent_futures.py

```diff
@@ -342,6 +342,29 @@ def test_hang_issue12364(self):
         for f in fs:
             f.result()
 
+    def test_cancel_futures(self):
+        executor = self.executor_type()
+        fs = [executor.submit(time.sleep, .1) for _ in range(50)]
+        executor.shutdown(cancel_futures=True)
+        # We can't guarantee the exact number of cancellations, but we can
+        # guarantee that *some* were cancelled. In this case, at least half
+        # should have been cancelled.
+        cancelled = [fut for fut in fs if fut.cancelled()]
+        self.assertTrue(len(cancelled) >= 25, msg=f"{len(cancelled)=}")
+
+        # Ensure the other futures were able to finish.
+        # Use "not fut.cancelled()" instead of "fut.done()" to include futures
+        # that may have been left in a pending state.
```

Comment:
Hmm, how can futures be left in a pending state if you gave `wait=True`? (which is the default)

Reply:
There was a previous (flawed) version of one of my implementations of `cancel_futures` for PPE that resulted in running futures being left in a pending state, despite `wait=True`, so I wanted to test for it explicitly. If you're curious about the details, see https://github.com/python/cpython/pull/18057#discussion_r370899912. 

If you and @brianquinlan prefer though, I could remove this test. I see it as more of a "sanity check" than anything at this point, which I've found can often yield surprising results when debugging concurrency issues.

Reply:
Ah, sorry, I misunderstood the intent of the test (I had overlooked `self.assertTrue(fut.done())` below).
It's fine as written now.